storagenode/piecestore/verification.go

```diff
@@ -104,18 +92,18 @@ func (endpoint *Endpoint) VerifyOrder(ctx context.Context, peer *identity.PeerId
 		return ErrProtocol.New("order exceeded allowed amount=%v, limit=%v", order.Amount, limit.Limit) // TODO: report grpc status bad message
 	}
 
-	if err := signing.VerifyOrderSignature(ctx, signing.SigneeFromPeerIdentity(peer), order); err != nil {
-		return ErrVerifyUntrusted.New("invalid order signature") // TODO: report grpc status bad message
+	if err := signing.VerifyUplinkOrderSignature(ctx, limit.UplinkPublicKey, order); err != nil {
+		return ErrProtocol.Wrap(err)
```

Comment:
Why does this error case change from ErrVerifyUntrusted to ErrProtocol?

Reply:
fixed